axelar-chains-config/src/utils/sourcify.js

```
@@ -0,0 +1,140 @@
+        return [];
+    }
+}
+
+function searchContractName(rootFolder, targetValue) {
+    const files = fs.readdirSync(rootFolder);
+
+    for (const fileOrFolder of files) {
+        const fullPath = path.join(rootFolder, fileOrFolder);
+
+        if (fs.statSync(fullPath).isDirectory()) {
+            const contractName = searchContractName(fullPath, targetValue);
+
+            if (contractName) {
+                return contractName;
+            }
+        } else if (fileOrFolder.endsWith('.json')) {
+            const jsonData = JSON.parse(fs.readFileSync(fullPath, 'utf-8'));
+
+            if (jsonData.deployedBytecode === targetValue) {
+                return jsonData.contractName;
+            }
+        }
+    }
+}
+
+async function verify(dir, provider, address, chainId) {
+    try {
+        const bytecode = await provider.getCode(address);
+
+        if (!dir) {
+            throw new Error('Need to specify directory to verify on sourcify via solidity code');
+        }
+
+        const artifactsDir = path.join(__dirname, `../../../${dir}/artifacts/contracts`);
+        const contractName = await searchContractName(artifactsDir, bytecode);
+
+        if (!contractName) throw new Error('Byte code match not found');
+
+        const directoryPath = artifactsDir.substring(0, artifactsDir.length - 'contracts'.length) + 'build-info';
+        const res = await processDirectory(directoryPath, contractName);
+
+        if (res.length !== 2) throw new Error('Unable to find require metadata and solidity code for sourcify verification');
+        const [sol, metadata] = res;
+
+        await uploadToSourcify(metadata, sol, address, chainId);
+        console.log('Verified on Sourcify via uplaoding solidity code and metadata');
+    } catch (error) {
+        console.error('Unable to verify on Sourcify by uploading solidity code and metadata: ', error.message);
+
+        try {
+            await verifyFromEtherscan(chainId, address);
```

only need to catch the error for uploading to sourcify and do this as a fallback. But if this fails, we should propagate the failure to the caller